shared-libs/lineage/test/integration.js

```diff
@@ -0,0 +1,535 @@
+  }).then(data => {
+    const docs = data.rows.map(row => {
+      const doc = row.doc;
+      if (doc) {
+        doc._deleted = true;
+        return doc;
+      }
+    }).filter(doc => doc);
+    return db.bulkDocs(docs);
+  });
+};
+
+describe('Lineage', function() {
+  let lineage;
+
+  before(function() {
+    return memdownMedic('../../webapp/src')
+      .then(database => {
+        db = database;
+        lineage = lineageFactory(Promise, db);
+        return db.bulkDocs(fixtures);
+      });
+  });
+
+  after(function() {
+    const docIds = fixtures.map(doc => doc._id);
+    return deleteDocs(docIds);
+  });
+
+  afterEach(function() {
+    sinon.restore();
+  });
+
+  describe('fetchLineageById', function() {
+    it('returns correct lineage', function() {
+      return lineage.fetchLineageById(one_parent._id).then(result => {
+        chai.expect(result).to.have.lengthOf(2);
+        chai.expect(result[0]).excluding('_rev').to.deep.equal(one_parent);
+        chai.expect(result[1]).excluding('_rev').to.deep.equal(dummyDoc);
+      });
+    });
+  });
+
+  describe('fetchHydratedDoc', function() {
+    it('returns errors from query', function() {
+      return lineage.fetchHydratedDoc('abc').then(
+        () => {
+          throw new Error('should have thrown a 404');
+        },
+        err => {
+          chai.expect(err.status).to.equal(404);
+        }
+      );
+    });
+
+    it('returns unmodified doc when there is no lineage and no contact', function() {
+      return lineage.fetchHydratedDoc('no_lineageContact').then(actual => {
+        chai.expect(actual).excluding('_rev').to.deep.equal(no_lineageContact);
+      });
+    });
+
+    it('handles doc with unknown parent by leaving just the stub', function() {
+      return lineage.fetchHydratedDoc(stub_parents._id).then(actual => {
+        chai.expect(actual).excludingEvery('_rev').to.deep.equal({
+          _id: stub_parents._id,
+          type: stub_parents.type,
+          contact: { _id: 'something' },
+          parent: {
+            _id: 'something_else',
+            parent: dummyDoc
+          }
+        });
+      });
+    });
+
+    it('handles doc with unknown contact by leaving just the stub', function() {
+      return lineage.fetchHydratedDoc(stub_contacts._id).then(actual => {
+        chai.expect(actual).excludingEvery('_rev').to.deep.equal({
+          _id: stub_contacts._id,
+          type: stub_contacts.type,
+          contact: {
+            _id: 'something',
+            parent: dummyDoc
+          },
+          form: {}
+        });
+      });
+    });
+
+    it('handles missing contacts', function() {
+      return lineage.fetchHydratedDoc(no_contact._id).then(actual => {
+        chai.expect(actual).excluding('_rev').to.deep.equal(no_contact);
+      });
+    });
+
+    it('attaches the full lineage for reports', function() {
+      return lineage.fetchHydratedDoc(report._id).then(actual => {
+        chai.expect(actual).to.shallowDeepEqual({
```

Comment:
`shallowDeepEqual`?!  From http://www.chaijs.com/plugins/chai-shallow-deep-equal/, it sounds like this is checking if the actual result is a superset of the expected result.